test/data/livechat/ILiveChatRepository.test.ts

```diff
@@ -0,0 +1,383 @@
+import { ILivechatRoom, IVisitor } from '@rocket.chat/apps-engine/definition/livechat';
+import { IMessageAttachment } from '@rocket.chat/apps-engine/definition/messages';
+import { assert } from 'chai';
+import { anything, capture, instance, mock, verify, when } from 'ts-mockito';
+
+import ILiveChatCacheDataSource from '../../../app/data/livechat/cache-strategy/ILiveChatCacheDataSource';
+import ILiveChatInternalDataSource from '../../../app/data/livechat/cache-strategy/ILiveChatInternalDataSource';
+import LiveChatCacheStrategyRepositoryImpl from '../../../app/data/livechat/cache-strategy/LiveChatCacheStrategyRepositoryImpl';
+import ILiveChatRepository from '../../../app/data/livechat/ILiveChatRepository';
+import Department from '../../../app/domain/Department';
+import Room from '../../../app/domain/Room';
+import departmentFactory from '../../factories/DepartmentFactory';
+import livechatRoomFactory from '../../factories/LivechatRoomFactory';
+import roomFactory from '../../factories/RoomFactory';
+import visitorFactory from '../../factories/VisitorFactory';
+
+describe('ILiveChatRepository', () => {
+
+    let mockedCache: ILiveChatCacheDataSource = mock<ILiveChatCacheDataSource>();
+    let mockedInternal: ILiveChatInternalDataSource = mock<ILiveChatInternalDataSource>();
```

Comment:
As these mock objects are instantiated on `beforeEach`, you can declare them only with the types.